lib/ldp.js

```diff
@@ -266,11 +266,15 @@ class LDP {
     var root = !ldp.idp ? ldp.root : ldp.root + host + '/'
     var filePath = utils.uriToFilename(resourcePath, root, host)
 
-    // PUT requests not supported on containers. Use POST instead
-    if (filePath.endsWith('/')) {
-      return callback(error(409,
-        'PUT not supported on containers, use POST instead'))
+    const statusCode = fs.existsSync(filePath) ? 204 : 201
```

Comment:
If you prefer. I used a sync operation here because `exists` is a pretty lightweight operation, and it avoids introducing another level of callback nesting to the function. Happy to change it tho.